src/aklite_client_ext.cc

```diff
@@ -26,25 +27,28 @@
 // Returns the target that should be installed, if any.
 // It might be an updated version, a rollback target, or even the currently installed target, in case we need to sync
 // apps
-GetTargetToInstallResult AkliteClientExt::GetTargetToInstall(const LocalUpdateSource* local_update_source, int version,
+GetTargetToInstallResult AkliteClientExt::GetTargetToInstall(const CheckInResult& checkin_res, int version,
                                                              const std::string& target_name, bool allow_bad_target,
-                                                             bool force_apps_sync) {
-  // Tell CheckIn and CheckInLocal methods to not call notifyTufUpdateFinished in case of success
-  usingUpdateClientApi = true;
-  client_->setAppsNotChecked();
-  bool rollback_operation = false;
-
-  CheckInResult checkin_res = local_update_source == nullptr ? CheckIn() : CheckInLocal(local_update_source);
-
+                                                             bool force_apps_sync, bool is_offline_mode) {
+  std::string err;
   if (!checkin_res) {
-    LOG_WARNING << "Unable to update latest metadata";
-    return GetTargetToInstallResult(checkin_res);
+    err = "Can't select target to install using a failed check-in result";
+    LOG_WARNING << err << " " << static_cast<int>(checkin_res.status);
+    if (!invoke_post_cb_at_checkin_) {
+      client_->notifyTufUpdateFinished(err);
+    }
+    return {GetTargetToInstallResult::Status::Failed, TufTarget(), err};
   }
 
+  client_->setAppsNotChecked();
```

Comment:
My understanding is that `setAppsNotChecked` is not needed if the condition at line 34 is true (negative checkin result)?

Reply:
@detsch Have you checked this? It looks like `setAppsNotChecked` should be invoked if if (!checkin_res) {` is true too?